language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,12 +188,15 @@ protected boolean hasReadonlyKW(NonTerminalNode evalNodeAtCursor) {
         return this.expressionCompletions(context);
     }
 
-    protected NonTerminalNode getEvalNode(BallerinaCompletionContext context) {
+    protected Optional<Node> getEvalNode(BallerinaCompletionContext context) {
         NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
-        NonTerminalNode evalNode = (nodeAtCursor.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE
-                || nodeAtCursor.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE)
-                ? nodeAtCursor.parent() : nodeAtCursor;
-        return evalNode;
+        Predicate<Node> predicate = node ->
+                node.kind() == SyntaxKind.MAPPING_CONSTRUCTOR
+                        || node.parent().kind() == SyntaxKind.MAPPING_CONSTRUCTOR
+                        || node.kind() == SyntaxKind.MAPPING_MATCH_PATTERN
+                        || node.parent().kind() == SyntaxKind.MAPPING_MATCH_PATTERN
+                        || (node.kind() == SyntaxKind.SPECIFIC_FIELD || node.kind() == SyntaxKind.COMPUTED_NAME_FIELD);
```

Comment:
Is there a specific reason to have parenthesis for `(node.kind() == SyntaxKind.SPECIFIC_FIELD || node.kind() == SyntaxKind.COMPUTED_NAME_FIELD)` else we can maintain the consistency

Reply:
Done